libqtile/window.py

```diff
@@ -1347,9 +1347,27 @@ def cmd_toggle_minimize(self):
 
     def cmd_bring_to_front(self):
         if self.floating:
-            self.window.configure(stackmode=StackMode.Above)
+            self.qtile.change_layer(self.window.wid)
+        else:
+            self._reconfigure_floating()  # automatically above
```

Comment:
If `self.qtile.change_layer(self.window.wid)` was called when the window is _not_ floating, would it have the same effect?

Reply:
It would still raise the window. But `_reconfigure_floating` also *makes* the window floating (which I find weird, but people might expect it by now).

Reply:
If `bring_front_click` is `True`, it shows a different behaviour, since it does **not** make windows floating. Maybe this would be a good point to unify that behaviour and just drop a few lines of code here?

Reply:
That seems like odd behaviour, but if that's the case, then fair enough. Maybe we should make that more explicit in the code by changing this to:
```
    def cmd_bring_to_front(self):
        self.qtile.change_layer(self.window.wid)
        self.floating = True
```

Reply:
I would like that, but calling `_reconfigure_floating` also could change the dimensions of the window.

Reply:
I was just trying to figure out some of the floating stuff, and it seems that the "proper" way to trigger floating is to use `_enablefloating`, which is used exclusively by some of the setters (floating, maximised, fullscreen). I think these setters are therefore the "proper" way to change those settings. `self.floating = True` uses the dimensions in `self._float_info`, which I reckon are the "proper" dimensions to use when triggering floating, so it seems that using `_reconfigure_floating` here to enable floating was an oversight. I'd vote for using `self.floating = True`, but interested to hear other comments in case I've misinterpreted things here.

Reply:
I've added it in an extra commit. Let's see how it fares.

Reply:
👍🏽